packages/server/src/fhir/operations/expand.ts

```diff
@@ -266,32 +277,70 @@ async function computeExpansion(
     } else {
       await includeInExpansion(include, expansion, codeSystem, params);
     }
-
-    if (expansion.length > (count ?? MAX_EXPANSION_SIZE)) {
-      // Return partial expansion
-      break;
-    }
   }
 
+  await addExpandJob(valueSet, codeSystemCache); // Precompute expansion in the background for faster future searches
   return expansion;
 }
 
+async function isExpansionPrecomputed(valueSet: ValueSet): Promise<boolean> {
+  const result = await new SelectQuery('ValueSet_Membership')
```

Comment:
👍 

As discussed offline, this won't handle a couple of cases:
1. The "zero expanded values" case - where we would end up repeatedly trying the expansion
    * Probably fine, because (1) that's a rare event and (2) those are probably fast operations
    * But still an edge case that we may want to better handle in the future
2. The "expansion in progress" case - where partial results are in the table
    * Probably fine because (1) only applies during small time window and (2) partial results are not catastrophic

We can perhaps file an issue for a future `ValueSet_State` table that tracks all of this explicitly, or whatever other clever solutions you have in mind.